// n8n-nodes-pdfco/dist/credentials/PdfcoApi.credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PdfcoApi = void 0;
const constants_1 = require("../nodes/PdfCo/constants");
class PdfcoApi {
    constructor() {
        this.name = 'pdfcoApi';
        this.displayName = 'PDF.co API';
        this.properties = [
            {
                displayName: 'Enter your PDF.co API Key',
                name: 'apiKey',
                type: 'string',
                default: '',
                typeOptions: {
                    password: true,
                },
                hint: `To get your PDF.co API key please <a href="https://app.pdf.co/signup?utm_source=n8n&utm_medium=sign-up">click here to create your account</a>`,
            }
        ];
        this.authenticate = {
            type: 'generic',
            properties: {
                headers: {
                    'x-api-key': '={{$credentials?.apiKey}}',
                    'user-agent': constants_1.PDFCO_CONSTANTS.USER_AGENT,
                },
            },
        };
        this.test = {
            request: {
                url: `${constants_1.PDFCO_CONSTANTS.BASE_URL}/v1/account/credit/balance`,
            },
        };
    }
}
exports.PdfcoApi = PdfcoApi;
