// n8n-nodes-pdfco/dist/nodes/PdfCo/actions/mergePdf/description.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mergePdfDescription = void 0;
exports.mergePdfDescription = [
    {
        displayName: 'Input Link(s)',
        name: 'url',
        type: 'string',
        required: true,
        default: [],
        placeholder: 'https://example.com/invoice.pdf',
        typeOptions: {
            multipleValues: true,
            multipleValueButtonText: 'Add URL',
        },
        displayOptions: {
            show: {
                operation: ['Merge PDF'],
            },
        },
    },
    {
        displayName: 'Automatically Convert Non-PDF Files',
        name: 'autoConvert',
        type: 'boolean',
        default: false,
        displayOptions: {
            show: {
                operation: ['Merge PDF'],
            },
        },
    },
    {
        displayName: 'Advanced Options',
        name: 'advancedOptions',
        type: 'collection',
        placeholder: 'Add Option',
        default: {},
        displayOptions: {
            show: {
                operation: ['Merge PDF'],
            },
        },
        options: [
            {
                displayName: 'File Name',
                name: 'name',
                type: 'string',
                default: '',
            },
            {
                displayName: 'Webhook URL',
                name: 'callback',
                type: 'string',
                default: '',
                placeholder: 'https://example.com/callback',
            },
            {
                displayName: 'Custom Profiles',
                name: 'profiles',
                type: 'string',
                default: '',
                placeholder: `{ 'outputDataFormat': 'base64' }`,
            },
        ],
    },
    {
        displayName: 'Refer to our guide at https://developer.pdf.co/integrations/zapier/merge/index.html for details.',
        name: 'documentationDisplay',
        type: 'notice',
        default: '',
        displayOptions: {
            show: {
                operation: ['Merge PDF'],
            },
        },
    },
];
