// n8n-nodes-pdfco/dist/nodes/PdfCo/actions/aiInvoiceParser.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
exports.execute = execute;
const GenericFunctions_1 = require("../GenericFunctions");
exports.description = [
    {
        displayName: 'PDF URL',
        name: 'url',
        type: 'string',
        required: true,
        default: '',
        placeholder: 'https://example.com/invoice.pdf',
        description: 'URL to the source PDF file',
        hint: `Enter the URL of the invoice you want to extract. Make sure it's accessible and contains only one invoice. Use our <b>PDF Split</b> module if it includes multiple invoices.`,
        displayOptions: {
            show: {
                operation: [GenericFunctions_1.ActionConstants.AiInvoiceParser],
            },
        }
    },
    {
        displayName: 'Custom Fields',
        name: 'customfield',
        type: 'string',
        default: '',
        placeholder: 'storeNumber, lineTotal, financialCharges',
        hint: `Extract fields not included in the <a href="https://docs.pdf.co/api-reference/ai-invoice-parser#invoice-schema">default list</a> by entering fields in comma-separated format (e.g., <em>storeNumber, lineTotal, financialCharges</em>)`,
        displayOptions: {
            show: {
                operation: [GenericFunctions_1.ActionConstants.AiInvoiceParser],
            },
        }
    },
    {
        displayName: 'Advanced Options',
        name: 'advancedOptions',
        type: 'collection',
        placeholder: 'Add Option',
        default: {},
        displayOptions: {
            show: {
                operation: [GenericFunctions_1.ActionConstants.AiInvoiceParser],
            },
        },
        options: [
            {
                displayName: 'Webhook URL',
                name: 'callback',
                type: 'string',
                default: '',
                description: 'The callback URL or Webhook used to receive the output data',
                placeholder: 'https://example.com/callback',
                hint: 'Enter a webhook URL to receive the extracted data automatically when parsing is done.',
            },
        ],
    }
];
async function execute(index) {
    const inputUrl = this.getNodeParameter('url', index);
    const customfield = this.getNodeParameter('customfield', index);
    const advancedOptions = this.getNodeParameter('advancedOptions', index);
    const callback = advancedOptions === null || advancedOptions === void 0 ? void 0 : advancedOptions.callback;
    const endpoint = `/v1/ai-invoice-parser`;
    const body = { url: inputUrl, customfield: customfield, async: true };
    if (callback)
        body.callback = callback;
    const responseData = await GenericFunctions_1.pdfcoApiRequestWithJobCheck.call(this, endpoint, body);
    return this.helpers.returnJsonArray(responseData);
}
