// n8n-nodes-pdfco/dist/nodes/PdfCo/actions/searchPdf.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
exports.execute = execute;
const GenericFunctions_1 = require("../GenericFunctions");
exports.description = [
    {
        displayName: 'PDF URL',
        name: 'url',
        type: 'string',
        required: true,
        default: '',
        placeholder: 'https://example.com/document.pdf',
        description: 'The URL of the PDF file to search',
        hint: 'Source file URL of the PDF file to search',
        displayOptions: {
            show: {
                operation: [GenericFunctions_1.ActionConstants.SearchPdf],
            },
        },
    },
    {
        displayName: 'Search Query',
        name: 'searchString',
        type: 'string',
        required: true,
        default: '',
        description: 'Specify the text you wish to search for within the PDF document (e.g. company name)',
        placeholder: 'e.g.: company name',
        hint: 'Specify the text you wish to search for within the PDF document (e.g. company name)',
        displayOptions: {
            show: {
                operation: [GenericFunctions_1.ActionConstants.SearchPdf],
            },
        },
    },
    {
        displayName: 'Use Regular Expressions',
        name: 'regexSearch',
        type: 'boolean',
        default: false,
        description: 'Whether to use regular expressions for more complex search patterns',
        displayOptions: {
            show: {
                operation: [GenericFunctions_1.ActionConstants.SearchPdf],
            },
        },
    },
    {
        displayName: 'Pages',
        name: 'pages',
        type: 'string',
        default: '',
        description: 'Default: `0` (first page). Use ranges like `0,1-2,5,7-` (7- = from page 7 to end). Negative numbers count from end: `-2` = second-to-last page.',
        hint: 'Default: `0` (first page). Use ranges like `0,1-2,5,7-` (7- = from page 7 to end). Negative numbers count from end: `-2` = second-to-last page.',
        displayOptions: {
            show: {
                operation: [GenericFunctions_1.ActionConstants.SearchPdf],
            },
        },
    },
    {
        displayName: 'Advanced Options',
        name: 'advancedOptions',
        type: 'collection',
        placeholder: 'Add Option',
        default: {},
        displayOptions: {
            show: {
                operation: [GenericFunctions_1.ActionConstants.SearchPdf],
            },
        },
        options: [
            {
                displayName: 'File Name',
                name: 'name',
                type: 'string',
                default: '',
                description: 'Custom name for the output file. If empty, uses default file name.',
                hint: 'Custom name for the output file. If empty, uses default file name.',
            },
            {
                displayName: 'Webhook URL',
                name: 'callback',
                type: 'string',
                default: '',
                placeholder: 'https://example.com/callback',
                description: 'The callback URL or Webhook used to receive the output data',
                hint: `The callback URL or Webhook used to receive the output data`,
            },
            {
                displayName: 'Output Links Expiration (In Minutes)',
                name: 'expiration',
                type: 'number',
                default: 60,
                description: 'The expiration time of the output link in minutes',
            },
            {
                displayName: 'Word Matching Mode',
                name: 'wordMatchingMode',
                type: 'string',
                default: 'SmartMatch',
                options: [
                    {
                        name: 'Smart Match',
                        value: 'SmartMatch',
                    },
                    {
                        name: 'Exact Match',
                        value: 'ExactMatch',
                    },
                    {
                        name: 'None',
                        value: 'None',
                    },
                ],
                hint: 'Default: `SmartMatch`. Use `ExactMatch` to match the exact word. Use `None` to disable word matching.',
            },
            {
                displayName: 'Password',
                name: 'password',
                type: 'string',
                typeOptions: {
                    password: true,
                },
                default: '',
                description: 'The password of the PDF file',
                hint: 'The password of the password-protected PDF file',
            },
            {
                displayName: 'HTTP Username',
                name: 'httpusername',
                type: 'string',
                default: '',
                description: 'The HTTP username if required to access source URL',
                hint: `The HTTP username if required to access source URL`,
            },
            {
                displayName: 'HTTP Password',
                name: 'httppassword',
                type: 'string',
                typeOptions: {
                    password: true,
                },
                default: '',
                description: 'The HTTP password if required to access source URL',
                hint: `The HTTP password if required to access source URL`,
            },
            {
                displayName: 'Custom Profiles',
                name: 'profiles',
                type: 'string',
                default: '',
                placeholder: `{ 'outputDataFormat': 'base64' }`,
                description: 'Use "JSON" to adjust custom properties. Review Profiles at https://docs.pdf.co/api-reference/profiles/index.html to set extra options for API calls and may be specific to certain APIs.',
                hint: `Use "JSON" to adjust custom properties. Review <a href="https://docs.pdf.co/api-reference/profiles">Profiles documentation</a> to set extra options for API calls and may be specific to certain APIs.`,
            },
        ],
    },
];
async function execute(index) {
    const inputUrl = this.getNodeParameter('url', index);
    const searchString = this.getNodeParameter('searchString', index);
    const regexSearch = this.getNodeParameter('regexSearch', index);
    const pages = this.getNodeParameter('pages', index);
    const advancedOptions = this.getNodeParameter('advancedOptions', index);
    const body = {
        url: inputUrl,
        searchString,
        regexSearch,
        async: true,
        inline: true,
    };
    if (pages) {
        body.pages = pages;
    }
    const fileName = advancedOptions === null || advancedOptions === void 0 ? void 0 : advancedOptions.name;
    if (fileName)
        body.name = fileName;
    const callback = advancedOptions === null || advancedOptions === void 0 ? void 0 : advancedOptions.callback;
    if (callback)
        body.callback = callback;
    const expiration = advancedOptions === null || advancedOptions === void 0 ? void 0 : advancedOptions.expiration;
    if (expiration)
        body.expiration = expiration;
    const wordMatchingMode = advancedOptions === null || advancedOptions === void 0 ? void 0 : advancedOptions.wordMatchingMode;
    if (wordMatchingMode)
        body.wordMatchingMode = wordMatchingMode;
    const httpusername = advancedOptions === null || advancedOptions === void 0 ? void 0 : advancedOptions.httpusername;
    if (httpusername)
        body.httpusername = httpusername;
    const httppassword = advancedOptions === null || advancedOptions === void 0 ? void 0 : advancedOptions.httppassword;
    if (httppassword)
        body.httppassword = httppassword;
    const password = advancedOptions === null || advancedOptions === void 0 ? void 0 : advancedOptions.password;
    if (password)
        body.password = password;
    const profiles = advancedOptions === null || advancedOptions === void 0 ? void 0 : advancedOptions.profiles;
    if (profiles)
        body.profiles = profiles;
    (0, GenericFunctions_1.sanitizeProfiles)(body);
    const responseData = await GenericFunctions_1.pdfcoApiRequestWithJobCheck.call(this, '/v1/pdf/find', body);
    return this.helpers.returnJsonArray(responseData);
}
