// n8n-nodes-pdfco/dist/nodes/PdfCo/actions/versionDescription.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.versionDescription = void 0;
const aiInvoiceParse = __importStar(require("./aiInvoiceParser"));
const mergePdf = __importStar(require("./mergePdf"));
exports.versionDescription = {
    displayName: 'PDF.co Api',
    name: 'PDFco Api',
    description: 'Generate PDF, extract data from PDF, split PDF, merge PDF, convert PDF. Fill PDF forms, add text and images to pdf and much more with pdf.co!',
    defaults: {
        name: 'PDFco Api',
    },
    group: ['transform'],
    icon: 'file:pdfco.svg',
    inputs: ['main'],
    outputs: ['main'],
    credentials: [
        {
            name: 'pdfcoApi',
            required: true,
        },
    ],
    properties: [
        {
            displayName: 'Actions',
            name: 'operation',
            type: 'options',
            noDataExpression: true,
            options: [
                {
                    name: 'AI Invoice Parser',
                    value: 'AI Invoice Parser',
                    description: 'Transform your invoice processing with our advanced AI. Enjoy quick, accurate data extraction that eliminates manual effort and saves you valuable time.',
                    action: 'AI Invoice Parser',
                },
                {
                    name: 'Merge PDF',
                    value: 'Merge PDF',
                    description: 'Combine multiple document formats into a single PDF. Supported formats include PDF, DOC, DOCX, XLS, JPG, PNG, and more.',
                    action: 'Merge PDF',
                },
            ],
            default: 'AI Invoice Parser',
        },
        ...aiInvoiceParse.description,
        ...mergePdf.description,
    ],
    subtitle: '={{$parameter["operation"]}}',
    version: 1,
};
