// n8n-nodes-pdfco/dist/nodes/PdfCo/actions/extraction/aiInvoiceParser/description.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.aiInvoiceParserDescription = void 0;
exports.aiInvoiceParserDescription = [
    {
        displayName: 'Url',
        name: 'url',
        type: 'string',
        required: true,
        default: '',
        placeholder: 'https://example.com/invoice.pdf',
        displayOptions: {
            show: {
                operation: ['aiInvoiceParser'],
                resource: ['extraction'],
            },
        }
    },
    {
        displayName: 'Advanced Options',
        name: 'advancedOptions',
        type: 'collection',
        placeholder: 'Add Option',
        default: {},
        displayOptions: {
            show: {
                operation: ['aiInvoiceParser'],
                resource: ['extraction'],
            },
        },
        options: [
            {
                displayName: 'Webhook URL',
                name: 'callback',
                type: 'string',
                default: '',
                placeholder: 'https://example.com/callback',
            },
        ],
    },
    {
        displayName: 'Refer to our guide at https://developer.pdf.co/integrations/zapier/ai-invoice-parser/index.html for details.',
        name: 'documentationDisplay',
        type: 'notice',
        default: '',
        displayOptions: {
            show: {
                operation: ['aiInvoiceParser'],
                resource: ['extraction'],
            },
        },
    },
];
